include/logger/logger_lib.hpp

```diff
@@ -3,7 +3,62 @@
 
 #include "sink.hpp"
 
+/**
+ * @brief Macro FLAG_ON() extracts the log data such as date, time, line number,file name,function name from where it is being
```

Comment:
is it mandatory for user to call this along with log function call ? 

Why name FLAG_ON() you can use META_DATA or some other Name

Reply:
Yes it is mandatory for the user to pass this argument while logging , 
example obj.trace(" x=10", FLAG_ON());   , by this means meta data will be captured . 

since internally  it is a metadata ,but  at user side it is like a flag  hence named flag .